opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -526,6 +566,11 @@ message HistogramDataPoint {
   // 1970.
   fixed64 time_unix_nano = 3;
 
+  // Field "flags" is used as a bit-field as defined in enum DataPointFlags.
+  // If FLAG_NO_RECORDED_VALUE is set, the "count", "sum", "bucket_counts",
+  // "explicit_bounds" fields are undefined.
+  uint32 flags = 10;
```

Comment:
I think uint32 packed between two fixed64 will result in wasted 4 bytes in memory due to alignment rules in most (?) compiled languages. Probably better to move this to the end. It won't help with space usage right now (since the last field is also 64bit-aligned and the entire struct is likely too large and most heap allocators will pad it at least to 64bit boundary anyway), but in the future if we add any other field with 32bit or smaller alignment size they will get packed better together.

Reply:
I think protobuf packs based on ids, and thus flags is at the end at the moment (id = 10).  Although in the middle (for logical organization) in source code.

Reply:
Humm, I guess not.

```
When a message is serialized, there is no guaranteed order for how its known or unknown fields should be written. Serialization order is an implementation detail and the details of any particular implementation may change in the future. Therefore, protocol buffer parsers must be able to parse fields in any order.
```

I also don't know if the order in the source code dictates the order of packing. Given serialization is implementation details, I don't know we should worry about this as I don't know how to guarantee this between different protobuf implementations.

Reply:
Sorry, I was not clear. I am talking about in-memory representation. At least in Go the in-memory representation is a struct with the field listed in the same order they are listed in the proto (regardless of the id numbers). Due to struct field alignment rules this will result in an unused hole of 4 bytes between `flags` and `count` fields.

Reply:
I see.